src/service/resources-impl.js

```diff
@@ -344,6 +344,7 @@ export class Resources {
     const resource = new Resource((++this.resourceIdCounter_), element, this);
     if (!element.id) {
       element.id = 'AMP_' + resource.getId();
+      element.setAttribute('amp-auto-id', element.id);
```

Comment:
Another request. May be we should also expose this `resource.getId()` via a more direct API on custom AMP element? This would make reads _way_ faster than with `getAttribute`.